ecllibrary/std/Uni.ecl

```diff
@@ -398,4 +398,17 @@ EXPORT unicode GetNthWord(unicode text, unsigned4 n, varstring localename = '')
 EXPORT ExcludeNthWord(unicode text, unsigned4 n, varstring localename = '') :=
     lib_unicodelib.UnicodeLib.UnicodeLocaleExcludeNthWord(text, n, localename);
 
+/**
+ * Returns the source string with the all characters that match characters in the search string replaced
+ * with the character at the corresponding position in the replacement string.
```

Comment:
The UnicodeString member function that I used to count the code points is countChar32(), and the reference for it says "A code point may occupy either one or two char16_t code units. Counting code points involves reading all code units," suggesting that it might tell when a lone surrogate code unit has been involved.

I could try to implement compareCodePointOrder() instead, which explicitly mentions that the result is not defined if it contained unpaired surrogates.